code/assurance-testapp/build.gradle.kts

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import com.adobe.marketing.mobile.gradle.BuildConstants
+
+plugins {
+    id("com.android.application")
+    id("org.jetbrains.kotlin.android")
+}
+
+android {
+    namespace = "com.adobe.marketing.mobile.assurance.testapp"
+
+    defaultConfig {
+        applicationId = "com.adobe.mobile.marketing.assurance.testapp"
+        minSdk = BuildConstants.Versions.MIN_SDK_VERSION
+        compileSdk = BuildConstants.Versions.COMPILE_SDK_VERSION
+        targetSdk = BuildConstants.Versions.TARGET_SDK_VERSION
+
+        versionCode = 1
+        versionName = "1.0"
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+        vectorDrawables.useSupportLibrary = true
+    }
+
+    buildTypes {
+        getByName(BuildConstants.BuildTypes.RELEASE) {
+            isMinifyEnabled = false
+            proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
+        }
+    }
+
+    compileOptions {
+        sourceCompatibility = BuildConstants.Versions.JAVA_SOURCE_COMPATIBILITY
+        targetCompatibility = BuildConstants.Versions.JAVA_TARGET_COMPATIBILITY
+    }
+
+    kotlinOptions {
+        jvmTarget = BuildConstants.Versions.KOTLIN_JVM_TARGET
+        languageVersion = BuildConstants.Versions.KOTLIN_LANGUAGE_VERSION
+        apiVersion = BuildConstants.Versions.KOTLIN_API_VERSION
+    }
+
+    buildFeatures {
+        compose = true
+    }
+
+    composeOptions {
+        kotlinCompilerExtensionVersion = BuildConstants.Versions.COMPOSE_COMPILER
```

Comment:
I would recommend that the test app defines it own version of compose instead of accessing it from the BuildConstants.

Reply:
Discussed that due the way extension projects are setup to [use the Kotlin gradle plugin from the `aep-library`](https://github.com/adobe/aepsdk-commons/blob/dev/android/aepsdk-gradle-plugin/aepsdk-gradle-plugin/build.gradle.kts#L6
), we cannot use a version of Kotlin Compiler greater than what is defined there. When using a lower version, it should be compatible with the `composeVersion` being used. So decided to leave this as is and use a custom Compose version.